### DISCUSSION TITLE
Default language option for gatsby-remark-prismjs

### DISCUSSION THREAD

**Discussion by ben4d85** [gatsbyjs/gatsby#25615] [Ideas / Feature Requests]
## Summary

For [gatsby-remark-prismjs](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-prismjs), please consider providing an option to set a _default language_ in gatsby-config.js.

### Basic example

Right now, we have to specify the language manually every time we add code to a markdown file, for example:

```md
Lorem `csharp¬ipsum = dolor(sit);` amet.
```

### Motivation

For those of us using the same language almost exclusively, it would be much more convenient if there was an option to set a default language in gatsby-config.js, for example:

```js
{
    resolve: 'gatsby-remark-prismjs',
    options: {
        inlineCodeMarker: '¬',
        inlineCodeDefaultLanguage: 'csharp',
        blockCodeDefaultLanguage: 'csharp',
    },
},          
```

**Comment by pieh** [comment]
Would using additional plugin that would add default lang instead of having this baked in `gatsby-remark-prismjs` work for you? `gatsby-remark-prismjs` is already quite large "beast" and I would really like to avoid adding more options there

**Comment by pieh** [comment]
Here's `gatsby-remark-*` plugin that set's the default lang for code and inline code - https://www.npmjs.com/package/gatsby-remark-default-code-lang

**Comment by ben4d85** [comment]
Yes, that's what I had in mind! Thank you so much!

However, now that I am using it, I realised a few things:

1. This new plugin does not appear to work with [gatsby-remark-code-titles](https://github.com/DSchau/gatsby-remark-code-titles/)

I have tried 

> \```title=MyClass.cs
> (...)
> \```

and 

> \```:title=MyClass.cs
> (...)
> \```

but both lead to the resulting code-block having no language at all (i.e. not using the default language configured in the new plugin).

2. If you are using VSCode to edit your markdown, then not specifying the language manually means you obviously are not going to get syntax highlighting for that language. Nothing to do with the plugin of course, but just a useful aside.

**Comment by pieh** [comment]
Ah, I didn't account for extra "directives" there. You can check the code of the plugin - https://github.com/pieh/gatsby-remark-default-code-lang/blob/master/index.js in both `title=MyClass.cs` and `:title=MyClass.cs` the `if (!node.lang)` check will not match, because `lang` will be those strings, so there would be need for some processing there, will adjust

**Comment by pieh** [comment]
Ok, added compatibility for `gatsby-remark-code-titles` in `0.0.2` - check updated readme for plugin ordering in config https://www.npmjs.com/package/gatsby-remark-default-code-lang
